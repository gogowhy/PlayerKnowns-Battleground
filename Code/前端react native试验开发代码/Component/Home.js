import React, { Component } from 'react';
import {
    TouchableOpacity,
    Text,
    ImageBackground,
    View,
    StyleSheet,
} from 'react-native';

export default class Home extends Component {

    constructor(props){
        super(props);
        this.ToLogin = this.ToLogin.bind(this);
    }

    /**
     * 跳转到登陆界面
     */
    ToLogin() {
        const { navigate } = this.props.navigation;  //获取navigation的navigate方法
        navigate('Login');  //跳转到注册过的Login界面
    }
    /**
     * 渲染图形界面
     * @return {[type]} [返回所渲染的界面]
     */
    render() {
        return (
            <ImageBackground style={styles.background}
                source={require('../src/img/bg1.png')}>
                <TouchableOpacity
                    activeOpacity={1.0}  //设置背景被点击时，透明度不变
                    style={styles.container}>
                    <View style={styles.container}>
                        <Text style={styles.content}>PlayerKnown's</Text>
                        <Text style={styles.content}>BattleGround</Text>
                    </View>
                    <TouchableOpacity
                        onPress={this.ToLogin}  //添加点击事件
                        style={styles.button}>
                        <Text
                            style={styles.btText}>开始游戏</Text>
                    </TouchableOpacity>
                </TouchableOpacity>
            </ImageBackground>
        );
    }

}

/**
 * 设置界面的布局样式
 * @type {[StyleSheet]}
 */
const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        //backgroundColor: '#F5FCFF',
    },
    content: {
        fontFamily: 'Calibri',
        fontSize: 40,
        color: '#000',
        fontWeight: 'bold',
        marginTop: 10,
        marginBottom: 10,
    },
    background: {
        position: 'absolute',
        left: 0,
        right: 0,
        top: 0,
        bottom: 0,
        height: null,
        width: null,
        zIndex: -1,
    },
    button: {
        height: 40,
        width: 150,
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: 8,
        backgroundColor: '#FF4500',
        marginTop: 50,
        marginBottom: 70,
    },
    btText: {
        color: '#fff',
        fontWeight: 'bold',
        fontSize: 20,
    }
});