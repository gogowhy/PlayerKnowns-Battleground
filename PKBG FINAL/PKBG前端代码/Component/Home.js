import React, { Component } from 'react';
import {
    ImageBackground,
    View,
    Image,
    StyleSheet,
} from 'react-native';
import { Button } from 'native-base';
import base from '../src/style/base';
import BGmusic from './BGmusic';

export default class Home extends Component {

    constructor(props) {
        super(props);
        this.ToLogin = this.ToLogin.bind(this);
    }

    /**
     * 跳转到登陆界面
     */
    ToLogin() {
        const { navigate } = this.props.navigation;  //获取navigation的navigate方法
        navigate('Login');  //跳转到注册过的Login界面
    }
    /**
     * 渲染图形界面
     * @return {[type]} [返回所渲染的界面]
     */
    render() {
        return (
            <ImageBackground style={base.background}
                source={require('../src/img/ppp.png')}>
                <View style={base.container}>
                    <View style={{ height: 120 }} />
                    <Button
                        rounded
                        bordered
                        activeOpacity={0.5}
                        onPress={this.ToLogin} //-----------该属性需要保留！-------------
                        style={base.button}>
                        <Image
                            source={require('../src/img/start2.png')}
                            style={{ height: '180%', width: '180%', }}
                        />
                    </Button>
                </View>
                <BGmusic />
            </ImageBackground>
        );
    }
}
