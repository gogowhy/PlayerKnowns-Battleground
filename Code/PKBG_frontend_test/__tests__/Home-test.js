/**
 * @format
 */

import 'react-native';
import React from 'react';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import Home from '../Component/Home';

test('renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});
