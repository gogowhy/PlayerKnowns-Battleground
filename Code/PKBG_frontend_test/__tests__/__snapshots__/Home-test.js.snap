// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "bottom": 0,
      "height": null,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
      "width": null,
      "zIndex": -1,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../src/img/bg1.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "KustWoff",
            "fontSize": 50,
            "marginTop": 25,
          }
        }
      >
        PlayerKnown's
      </Text>
      <Text
        style={
          Object {
            "color": "#000",
            "fontFamily": "KustWoff",
            "fontSize": 50,
            "marginTop": 25,
          }
        }
      >
        BattleGround
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FF4500",
          "borderRadius": 8,
          "height": 40,
          "justifyContent": "center",
          "marginBottom": 70,
          "marginLeft": 10,
          "marginRight": 10,
          "marginTop": 50,
          "opacity": 1,
          "width": 150,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "zhenhunshoushu",
            "fontSize": 30,
          }
        }
      >
        开始游戏
      </Text>
    </View>
  </View>
</View>
`;
